### ISSUE TITLE
Negative running_var in frozen batchNorm

### ISSUE THREAD

**Issue by aabbas90** [facebookresearch/detectron2#2413]
## Description:

I am experiencing negative values in `running_var` parameter of FrozenBatchNorm in the ResNet backbone, which then leads to NaN values in forward pass. I think it is because of the following line:

https://github.com/facebookresearch/detectron2/blob/ebbc971ae976dec2d78942913c08da0eeee8a704/detectron2/layers/batch_norm.py#L85

# My workflow:
1. Took a pretrained backbone from detectron2 (detectron2://DeepLab/R-52.pkl).
2. Trained the backbone and my model from my custom task and save the checkpoint.
3. Note that BatchNorm from resulting checkpoint has version 2, but the checkpoint from step (1) had version 3.
4. Fine-tuned the outcome of step (2), by setting `FREEZE_AT=1`. Since my custom model has version 2, so the `-=eps` code cited above becomes effective and changes the `running_var`, which now contains negative values (and some of these values are equal to `self.eps`, so the variance effectively becomes 0). The older values of `running_var` are actually correct ones, so I would like to avoid this offset being applied.
5. As mentioned in (3), this problem does not happen when I do `FREEZE_AT=1` on the step (1), since that checkpoint has version 3.

# Questions:
1. Shall I need to explicitly set the version of checkpoint myself while saving so that my checkpoint does not have version 2?
2. Sorry, if some information is missing. I have a custom fork of detectron2 (but did not make any changes in checkpointing). But if still some information is missing, I can possibly collect. Thanks!

**Comment by ppwwyyxx** [comment]
> Note that BatchNorm from resulting checkpoint has version 2

why so? If it's because you want to use weights trained by __other__ layers into FrozenBatchNorm2d, then some conversion process is needed.

**Comment by aabbas90** [comment]
> > Note that BatchNorm from resulting checkpoint has version 2
> 
> why so? If it's because you want to use weights trained by **other** layers into FrozenBatchNorm2d, then some conversion process is needed.

So I just fine-tuned my model taken from step (2), without any changes except `FREEZE_AT=1`. I do not know what does a version of a layer mean,  I just looked at the attributes inside each layer and see that the value of this version is 2 which makes the variance negative. Thanks.

**Comment by ppwwyyxx** [comment]
yeah because `FrozenBatchNorm` is a different layer from `BatchNorm`, loading weights of one to another is not perfect unfortunately. The easiest manual workaround is to delete `_version` in the checkpoint (or set them to 3).

A more proper way could be: (1) load the original weights to the original model (without changing freeze settings) (2) call `convert_frozen_batchnorm` to freeze parts of the model (3) save the weights to a new file
